src/Compilers/CSharp/Portable/Symbols/TypeSymbol.cs

```diff
@@ -1114,17 +1141,7 @@ private static void ReportImplicitImplementationMatchDiagnostics(Symbol interfac
         private static void ReportImplicitImplementationMismatchDiagnostics(Symbol interfaceMember, TypeSymbol implementingType, Symbol closestMismatch, DiagnosticBag diagnostics)
         {
             // Determine  a better location for diagnostic squiggles.  Squiggle the interface rather than the class.
-            Location interfaceLocation = null;
-            if ((object)implementingType != null)
-            {
-                var @interface = interfaceMember.ContainingType;
-                SourceMemberContainerTypeSymbol snt = implementingType as SourceMemberContainerTypeSymbol;
-                interfaceLocation = snt.GetImplementsLocation(@interface) ?? implementingType.Locations[0];
-            }
-            else
-            {
-                interfaceLocation = implementingType.Locations[0];
-            }
+            Location interfaceLocation = GetInterfaceLocation(interfaceMember, implementingType);
```

Comment:
The code here previously handled `null` `implementingType`. But now it asserts that is it non-null.
I looked at the one caller for the `ReportImplicitImplementationMismatchDiagnostics`, but didn't see how it guarantees non-null `implementingType` here.